src/dashboard-refactor/lists-sidebar/components/lists-sidebar-item/index.tsx

```diff
@@ -0,0 +1,189 @@
+import React, { PureComponent } from 'react'
+import styled, { css, keyframes } from 'styled-components'
+
+import colors from '../../../colors'
+import { fonts } from '../../../styleConstants'
+
+import {
+    HoverState,
+    DroppableState,
+    SelectedState,
+    NewItemsCountState,
+} from 'src/dashboard-refactor/types'
+import { MoreActionButtonState } from './types'
+
+import Margin from 'src/dashboard-refactor/components/Margin'
+
+// probably want to use timing function to get this really looking good. This is just quick and dirty
+const blinkingAnimation = keyframes`
+    0% {
+        background-color: ${colors.onHover};
+    }
+    70% {
+        background-color: transparent;
+    }
+    100% {
+        background-color: ${colors.onHover};
+    }
+`
+
+const Container = styled.div`
+    height: 27px;
+    width: 100%;
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: center;
+    background-color: transparent;
+    ${(props) =>
+        props.isHovered &&
+        css`
+            background-color: ${colors.onHover};
+        `}
+    ${(props) =>
+        props.isHovered &&
+        props.isSelected &&
+        css`
+            background-color: ${colors.onSelect};
+        `}
+    ${(props) =>
+        props.isBlinking &&
+        css`
+            animation: ${blinkingAnimation} 0.4s 2;
+        `}
+    `
+
+const ListTitle = styled.div`
+    font-family: ${fonts.primary.name};
+    font-style: normal;
+    ${(props) =>
+        props.isSelected &&
+        css`
+            font-weight: ${fonts.primary.weight.bold};
+        `}
+    font-size: 14px;
+    line-height: 21px;
+    height: 18px;
+`
+
+const Icon = styled.div`
+    height: 12px;
+    width: 12px;
+    font-size: 12px;
+`
+
+const NewItemsCount = styled.div`
+    width: 30px;
+    height: 14px;
+    border-radius: 10px;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    background-color: ${colors.midGrey};
+    div {
+        font-family: ${fonts.primary.name};
+        font-weight: ${fonts.primary.weight.bold};
+        font-size: 10px;
+        line-height: 14px;
+    }
+`
+
+export interface ListsSidebarItemProps {
+    listName: string
+    isEditing: boolean
+    selectedState: SelectedState
+    hoverState: HoverState
+    droppableState: DroppableState
+    newItemsCountState: NewItemsCountState
+    moreActionButtonState: MoreActionButtonState
+}
+
+export default class ListsSidebarItem extends PureComponent<
+    ListsSidebarItemProps
+> {
+    private handleDragOver() {
```

Comment:
Is this indirection necessary? Can't we just call `this.props.*` direcly in the DOM event handlers?